### PR TITLE
MM-23646: Improve group sync performance.

### PR DESCRIPTION
**Pull request by mkraft**
#### Summary

New query to support an LDAP sync performance improvement.

See https://github.com/mattermost/enterprise/pull/600 for usage.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-23646

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -56,6 +56,7 @@ _testmain.go
 # Log files
 *.log
 *.log.jsonl
+*.log.gz
```
Comment (mkraft) [inline .gitignore]: Unrelated, just necessary for when large log files roll over during development.

**Comment by mkraft** [comment]
> Wish that opentracing_layer.go and timer_layer.go didn't duplicate so much code, but realize that is the pattern.

Those are both auto-generated with `make store-layers` and `make app-layers`.

**Comment by amyblais** [comment]
@mkraft Would this go to QA review next?

**Comment by mkraft** [comment]
Reverting b/c I accidentally merged this prior to QA review.

**Pull request merged**